chromadb/api/models/CollectionCommon.py

```diff
@@ -552,12 +565,10 @@ def _validate_and_prepare_delete_request(
     def _normalize_embeddings(
         embeddings: Union[  # type: ignore[type-arg]
             OneOrMany[Embedding],
-            OneOrMany[np.ndarray],
+            OneOrMany[PyEmbedding],
         ]
     ) -> Embeddings:
-        if isinstance(embeddings, np.ndarray):
-            return embeddings.tolist()  # type: ignore
-        return embeddings  # type: ignore
+        return cast(Embeddings, [np.array(embedding) for embedding in embeddings])
```

Comment:
question: do we validate the shape of PyEmbedding anywhere?

Reply:
Do you mean do we validate that the embeddings the user provide have the same dimensionality? If so, we do in `chromadb/api/segment.py` on line 828 in `_validate_dimension`.

Reply:
Right but do we also validate the length of the outer list as well.

Reply:
Yup, we make sure that it matches the length of ids in `_validate_embedding_set` in `chromadb/api/models/CollectionCommon.py`